File: tests/07_phases_ompss.dg/c/success_taskwait_depend_03.c
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
test_CFLAGS="--no-copy-deps"
</testinfo>
*/

#include<assert.h>

void foo()
{
    int x = 0, y = 2;

    #pragma omp task depend(inout: x) shared(x)
    x++;

    #pragma omp task depend(in: x) depend(inout: y) shared(x, y)
    y -= x;

    #pragma omp taskwait depend(inout: x)
    assert(x == 1);
    assert(x == y);
}

int main(int argc, char*argv)
{
    foo();
    return 0;
}

File: tests/07_phases_ompss.dg/fortran/success_taskwait_depend_03.f90
! <testinfo>
! test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
! test_FFLAGS="--no-copy-deps"
! </testinfo>

SUBROUTINE FOO()
    IMPLICIT NONE
    INTEGER :: X, Y

    X = 0
    Y = 2

    !$OMP TASK DEPEND(INOUT: X) SHARED(X)
        X = X + 1
    !$OMP END TASK

    !$OMP TASK DEPEND(IN: X) DEPEND(INOUT: Y) SHARED(X, Y)
        Y = Y - X
    !$OMP END TASK

    !$OMP TASKWAIT DEPEND(INOUT: X)
    IF (X /= 1) STOP -1
    IF (X /= Y) STOP -2
END SUBROUTINE FOO

PROGRAM P
    IMPLICIT NONE
    CALL FOO()
END PROGRAM P
